misc/toml-parser/src/main/java/io/ballerina/toml/validator/schema/SchemaDeserializer.java

```diff
@@ -0,0 +1,99 @@
+                                      JsonDeserializationContext jsonDeserializationContext)
+            throws JsonParseException {
+        JsonObject jsonObj = jsonElement.getAsJsonObject();
+        String type = jsonObj.get("type").getAsString();
+        switch (type) {
+            case "object":
+                return getObjectSchema(jsonDeserializationContext, jsonObj);
+            case "array":
+                return getArraySchema(jsonDeserializationContext, jsonObj);
+            case "integer":
+                return getNumericSchema(jsonObj, Type.INTEGER);
+            case "number":
+                return getNumericSchema(jsonObj, Type.NUMBER);
+            case "string":
+                return getStringSchema(jsonObj);
+            case "boolean":
+                return new BooleanSchema(Type.BOOLEAN);
+            default:
+                throw new JsonParseException("type " + type + " is not supported type");
+        }
+    }
+
+    private ObjectSchema getObjectSchema(JsonDeserializationContext jsonDeserializationContext, JsonObject jsonObj) {
+        JsonElement descriptionProperty = jsonObj.get("description");
+        String description = descriptionProperty != null ? descriptionProperty.getAsString() : null;
+        JsonElement additionalProperty = jsonObj.get("additionalProperties");
+        boolean additionalProperties = additionalProperty == null || additionalProperty.getAsBoolean();
+        JsonObject properties = jsonObj.get("properties").getAsJsonObject();
+        Set<Map.Entry<String, JsonElement>> entries = properties.entrySet();
+        Map<String, AbstractSchema> propertiesList = new HashMap<>();
+        for (Map.Entry<String, JsonElement> entry : entries) {
+            String key = entry.getKey();
+            AbstractSchema
+                    abstractSchema = jsonDeserializationContext.deserialize(entry.getValue(), AbstractSchema.class);
+            propertiesList.put(key, abstractSchema);
+        }
+        return new ObjectSchema(Type.OBJECT, description, additionalProperties, propertiesList);
+    }
+
+    private AbstractSchema getArraySchema(JsonDeserializationContext jsonDeserializationContext, JsonObject jsonObj) {
+        JsonElement items = jsonObj.get("items").getAsJsonObject();
+        AbstractSchema abstractSchema = jsonDeserializationContext.deserialize(items, AbstractSchema.class);
+        return new ArraySchema(Type.ARRAY, abstractSchema);
+    }
+
+    private StringSchema getStringSchema(JsonObject jsonObj) {
+        JsonElement patternProperty = jsonObj.get("pattern");
+        String pattern = patternProperty != null ? patternProperty.getAsString() : null;
+        return new StringSchema(Type.STRING, pattern);
+    }
+
+    private NumericSchema getNumericSchema(JsonObject jsonObj, Type type) {
+        JsonElement minimumProperty = jsonObj.get("minimum");
+        Double minimum = minimumProperty != null ? minimumProperty.getAsDouble() : null;
```

Comment:
Trying to get as double without checking the value first can lead to runtime errors, such as ClassCastExceptions, NumberFormatExceptions, etc. So we need to do the validations first, and throw proper errors.

Do we have any negative test cases for the schema?

Reply:
Ack. As the initial implementation I wasn't concerned with the schema validations. Anyways I've implemented validations for basic data types for now and test cases for them. We need to implement validations for required and optional fields as well. I will create a issue for this.

Reply:
#27039